config/routes.rb

```diff
@@ -662,6 +662,7 @@
     resources :badges, only: [:index]
     get "/badges/:id(/:slug)" => "badges#show", constraints: { format: /(json|html|rss)/ }
     resources :user_badges, only: [:index, :create, :destroy]
+    put "/user_badges/favorite" => "user_badges#favorite", constraints: { format: :json }
```

Comment:
Why dont we have the user badge id in the route itself here?

PUT /user_badges/:id/favorite

In any case toggling wont be REST as it's not indempotent and would require two different routes which are annoying to deal with client side...

Reply:
I honestly didn't think of it, but that does sound like a better idea.

Reply:
btw I looked at what we are about to do for pinning bookmarks and this is what the PR looks like so far for the route:

https://github.com/discourse/discourse/blob/b0fc6d089beb6207c6da2b887b0ad634120cf93e/config/routes.rb

I think we should follow this pattern

Reply:
Nice, that looks much better! Thanks for the pointer :)